### PR TITLE
Add playground to sparrowql

### PR DESCRIPTION
**Pull request by ErnestTeluk**
Add playground to sparrowql

### CODE REVIEW

File: playground/src/examples/example2.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    { from: 'employments', local: 'userId', to: 'employees', foreign: '_id'},
+];
+
+const start = 'employments';
+
+const projection = {
+    salonName: 'salons.name',
+    employee: 'employees.firstName',
+    employer: 'owners.firstName'
+  };`;
```
Comment (Monteth) [inline playground/src/examples/example2.ts]:
```suggestion
export const example2 = `const aliases = {
    'owners': 'users',
    'employees': 'users'
}

const relations = [
    { from: 'employments', local: 'restaurantId', to: 'restaurants', foreign: '_id'},
    { from: 'restaurants', local: 'businessId', to: 'businesses', foreign: '_id'},
    { from: 'businesses', local: 'ownerId', to: 'owners', foreign: '_id'},
    { from: 'employments', local: 'userId', to: 'employees', foreign: '_id'},
];

const start = 'employments';

const projection = {
    restaurantName: 'restaurants.name',
    employee: 'employees.name',
    employer: 'owners.name'
  };`;
```

File: playground/src/hooks/usePlayground.tsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+const fn = `
+  function get() {
+    if (typeof aliases !== 'undefined') {
+      return {projection, relations, start, aliases };
+    }
+    return { projection, relations, start };
+  }
+
+  get();
+`;
```
Comment (Monteth) [inline playground/src/hooks/usePlayground.tsx:15]: TODO: Add other parameters like limit, offset, and so on.

File: playground/src/hooks/usePlayground.tsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+  useEffect(() => {
+    const inputFromUrl = parseQuery();
+    if (inputFromUrl) {
+        setInput(inputFromUrl);
+    } else {
+        updateQuery(example1);
+    }
+  }, []);
+  const _setInput = (value: string) => {
```
Comment (Monteth) [inline playground/src/hooks/usePlayground.tsx]: To distinguish it from the state change, maybe let's change the name to something like `onInputChanged`, or change the state update to `setPlaygroundInputState`

File: playground/src/utils/index.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+export function updateQuery(value: string) {
+  try {
+    const serialized = JSON.stringify(value);
+    const compressed = compress(serialized);
+    const encoded = encodeURIComponent(compressed);
+    const hash = '?' + encoded;
+    window.location.hash = hash;
+  } catch (_) {
+    // It's alright.
```
Comment (Monteth) [inline playground/src/utils/index.ts:20]: Can we put a console error or warning?

**Review by Monteth — APPROVED** [review]

File: .github/workflows/CI.yml [hunk 5]
```diff
@@ -28,7 +28,7 @@ jobs:
         with:
           mongodb-version: ${{ matrix.mongodb-version }}
       - name: Install dependencies
-        run: npm i -g npm@8 --no-audit && npm ci --no-audit
+        run: npm i -g npm@8 --no-audit && npm ci --no-audit && cd playground && npm ci --no-audit
```
Comment (kestarumper) [inline .github/workflows/CI.yml]: [optional]
Maybe it's worth splitting it into separate phases?

**Review by Monteth — APPROVED** [review]

**Pull request merged**